src/scripts/main.js

```diff
@@ -1 +1,66 @@
 'use strict';
+
+document.addEventListener('contextmenu', (e) => {
+  e.preventDefault();
+});
+
+const promise1 = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+  });
+
+  setTimeout(() => reject(new Error('First promise was rejected')), 3000);
+});
+
+const promise2 = new Promise((resolve) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  });
+});
+
+let leftClick = false;
+let rightClick = false;
+
+const promise3 = new Promise((resolve, reject) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0) {
+      leftClick = true;
+    }
+
+    if (e.button === 2) {
+      rightClick = true;
+    }
+
+    if (leftClick && rightClick) {
```

Comment:
The third promise resolves when both left and right clicks are detected. Ensure that this logic aligns with the intended user interaction, as it might be challenging for users to trigger both clicks simultaneously.